app/models/region.rb

```diff
@@ -54,6 +54,23 @@ def log_payload
     attrs.symbolize_keys
   end
 
+  def self.sources
+    includes(:source).map(&:source)
+  end
+
+  def syncable_patients
+    Patient
+      .with_discarded
+      .where(registration_facility: facilities.sources)
+      .or(Patient
+            .with_discarded
+            .where(assigned_facility: facilities.sources))
+      .union(Patient
+               .with_discarded
+               .joins(:appointments)
+               .where(appointments: {facility: facilities.sources}))
```

Comment:
I'm confused about this query -- isn't this doing a lot more work than the previous query on Patient?  

```
  scope :syncable_to_region, ->(region) {
    with_discarded.where(registration_facility: region.facilities)
  }
```

We should keep the query the same if at all possible, there are enough other changes going on here at the same time. 😀

Reply:
Yeah, I think I might change the others to be this way also or conditionally enforce this query only for blocks.

We cannot launch block-level syncing without these additional changes (so I think it must be in this PR ideally), because the syncability is just not a wide enough net to be scalable. [ref](https://docs.google.com/document/d/1Cflct0Y-44IRUVw_5-NptcnNSX1UgAPBiXqoXHq22io/edit#heading=h.gegdriq7f9oe).